photon-api/src/main/scala/com/linkedin/photon/ml/data/LocalDataSet.scala

```diff
@@ -176,6 +217,107 @@ object LocalDataSet {
     result
   }
 
+  /**
+   * Compute feature ratio confidence interval lower bounds.
+   *
+   * @param labelAndFeatures An [[Array]] of (label, feature vector) tuples
+   * @param zScore The Z-score for the chosen two-tailed confidence level
+   * @param globalFeatureInstances The global instances with the features present
+   * @param globalPositiveInstances The global positive instances with the features present
+   * @param binaryIndices The binary feature columns indices
+   * @param epsilon The constant used to compute for extreme case of ratio modeling
+   * @return the lowerBounds for feature columns
+   */
+  protected[ml] def computeRatioCILowerBound(
+      labelAndFeatures: Array[(Double, Vector[Double])],
+      globalFeatureInstances: Array[Double],
+      globalPositiveInstances: Array[Double],
+      binaryIndices: Set[Int],
+      numFeatures: Int,
+      zScore: Double,
+      epsilon: Double = 0.5): Map[Int, Double] = {
```

Comment:
I don't think this should be a parameter, I think we should hardcode it as a constant in the companion object